aptible/resource_app.go

```diff
@@ -302,24 +301,46 @@ func validateServiceSizingPolicy(ctx context.Context, d *schema.ResourceDiff, _
 }
 
 func resourceAppCreate(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*providerMetadata).LegacyClient
-	envID := int64(d.Get("env_id").(int))
+	m := meta.(*providerMetadata)
+	client := m.Client
+	legacy := m.LegacyClient
+	ctx := m.APIContext(context.Background())
```

Comment:
usually this context parameter gets supplied as part of the function - see this: https://github.com/aptible/terraform-provider-aptible/blob/master/aptible/resource_database.go#L103

if you generate a new context from background, this might not behave consistently for different reasons. i think it's simply altering the signature of this function

<img width="1036" height="394" alt="image" src="https://github.com/user-attachments/assets/00103a57-39ab-45ac-9ff1-c5a033e642bf" />

it will present errors more cleanly with diagnostics as well as a result i believe

Reply:
i vibed this a bit just to see the delta but normally this is important instead of spawning new contexts from background because you generally pass that and use it. 

you typically spawn a new context from the background to generate a specific and more granular context, this is probably okay but it could cause issues